llvm/lib/Target/DirectX/DXILFlattenArrays.cpp

```diff
@@ -263,8 +263,13 @@ bool DXILFlattenArraysVisitor::visitGetElementPtrInst(GetElementPtrInst &GEP) {
   // merge the byte offsets. Otherwise, this GEP is itself the root of a GEP
   // chain and we need to deterine the root array type
   if (auto *PtrOpGEP = dyn_cast<GEPOperator>(PtrOperand)) {
-    assert(GEPChainInfoMap.contains(PtrOpGEP) &&
-           "Expected parent GEP to be visited before this GEP");
+
+    // If the parent GEP was not processed, then we do not want to process its
```

Comment:
The comment makes sense as does the fix, Was wondering should we have something more explict about type checking for structs and bailing?

Reply:
Agreed, I think the comment on 296 is a good spot to indicate that

Reply:
I could throw an error when there is an unsupported type, but would the DXILFlattenArrays pass really be the place to throw such errors?

Line 296 and below already has
```c++
    // If the root type is not an array, we don't need to do any flattening
    if (!isa<ArrayType>(RootTy))
      return false;
```
to indicate that only arrays are applicable to this transformation. 
The name of the pass "DXIL Flatten Arrays" to me also implies it only applies to arrays.

Reply:
This can be addressed in a follow-up PR if needed.